### DISCUSSION TITLE
How do I set customer trailers in a streaming response?

### DISCUSSION THREAD

**Discussion by mkmik** [hyperium/tonic#1101] [Q&A]
Is there a way to set customer trailers in a streaming response?

A quick scan through the sources seems to show that the `Body` implementation is done by `EncodeBody` whose `poll_trailers` implementation delegates to `EncodeState` which returns trailers only in case of errors and it doesn't allow to customize trailers.

Should I wrap my service with a custom tower layer and try to return custom trailers that way? Or is there a better approach?

**Comment by davidpdrsn** [comment]
The easiest way is probably to use https://docs.rs/hyper/0.14.20/hyper/struct.Body.html#method.channel

**Reply by starsoccer** [reply]
Any guide or documentation on how to do this exactly? I have a server side streaming request using grpc web and looking to also use trailers to include metadata like a unique request id number on each payload

**Reply by matthewmturner** [reply]
Im also curious how to do this - @davidpdrsn would you be able to share any more details on how that method can be used? Or, since its been a couple years, is there an alternative approach now?
